ktor-utils/src/main/kotlin/no/nav/dagpenger/ktor/client/metrics/PrometheusMetrics.kt

```diff
@@ -0,0 +1,75 @@
+package no.nav.dagpenger.ktor.client.metrics
+
+import io.ktor.client.HttpClient
+import io.ktor.client.call.HttpClientCall
+import io.ktor.client.features.HttpClientFeature
+import io.ktor.client.request.HttpRequestBuilder
+import io.ktor.client.request.HttpSendPipeline
+import io.ktor.client.statement.HttpReceivePipeline
+import io.ktor.util.AttributeKey
+import io.ktor.util.pipeline.PipelinePhase
+import io.prometheus.client.CollectorRegistry
+import io.prometheus.client.Counter
+import io.prometheus.client.Histogram
+
+internal class PrometheusMetrics(registry: CollectorRegistry, baseName: String) {
+    private val duration = Histogram
+        .build("duration", "Time spent in requests")
+        .namespace(baseName)
+        .register(registry)
+
+    private val httpStatus = Counter
+        .build("status", "Count status codes for responses")
+        .namespace(baseName)
+        .labelNames("status")
+        .register(registry)
+
+    class Config {
+        var baseName: String = "ktor_client_metrics"
+        var registry: CollectorRegistry = CollectorRegistry.defaultRegistry
+    }
+
+    companion object Feature : HttpClientFeature<Config, PrometheusMetrics> {
+        override val key = AttributeKey<PrometheusMetrics>("metrics")
+
+        override fun prepare(block: Config.() -> Unit): PrometheusMetrics {
+            val conf = Config().apply(block)
+            return PrometheusMetrics(
+                registry = conf.registry,
```

Comment:
Jeg prøvde å lage støtte for å sende inn eget CollectorRegistry for å unngå å bruke `defaultRegistry` i testene, men det funker bare ikke. Den ender alltid opp med `defaultRegistry` uansett.